src/commonMain/kotlin/com/charleskorn/kaml/YamlNode.kt

```diff
@@ -63,6 +63,9 @@ public data class YamlScalar(val content: String, override val path: YamlPath) :
                     content.toFloat()
                 } catch (e: NumberFormatException) {
                     throw YamlScalarFormatException("Value '$content' is not a valid floating point value.", path, content)
+                } catch (e: IndexOutOfBoundsException) {
+                    // https://youtrack.jetbrains.com/issue/KT-69327
```

Comment:
Perhaps this would be clearer?
```suggestion
                    // Workaround for https://youtrack.jetbrains.com/issue/KT-69327
```

Could you please make the same change below as well?